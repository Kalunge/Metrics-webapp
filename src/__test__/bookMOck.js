const book = {
  name: 'myBook',
  chapters: 5,
};

export const match = {
  params: {
    abbrev: 'ex',
  },
};

export default book;
