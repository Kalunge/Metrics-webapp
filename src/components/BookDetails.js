import React from 'react';
import { FaGlobeEurope } from 'react-icons/fa';
import ChapterDetails from './ChapterDetails';

const BookDetails = () => (
  <div className="row">
    <div className="col-7">
      <FaGlobeEurope size={98} className="fa-icon" />
    </div>
    <div className="col-5 mt-5 text">
      <h1 className="d-block">Leviticus </h1>
      <p className="btn btn-dark">35 Chapters</p>
    </div>
    <h5 className="text mt-2">Chapter Breakdown</h5>
    <ChapterDetails />
  </div>
);

export default BookDetails;
