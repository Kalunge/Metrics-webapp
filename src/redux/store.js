import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import booksReducer from './books/books';

const middleware = [thunk, logger];

const reducer = combineReducers({
  books: booksReducer,
});
const store = createStore(reducer, applyMiddleware(...middleware));

export default store;
