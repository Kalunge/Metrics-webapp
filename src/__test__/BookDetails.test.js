import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import BookDetails from '../components/BookDetails';
import store from '../redux/store';
import { match } from './bookMOck';

it('renders   BookDetails correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BookDetails match={match} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
