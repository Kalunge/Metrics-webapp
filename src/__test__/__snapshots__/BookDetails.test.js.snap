// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders   BookDetails correctly 1`] = `
Array [
  <div
    className="navBar mt-3 mb-3"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <svg
        color="pink"
        fill="currentColor"
        height={42}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "pink",
          }
        }
        viewBox="0 0 512 512"
        width={42}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
        />
      </svg>
    </button>
    <h2 />
    <svg
      color="pink"
      fill="currentColor"
      height={42}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "pink",
        }
      }
      viewBox="0 0 24 24"
      width={42}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
      />
    </svg>
    <svg
      color="pink"
      fill="currentColor"
      height={42}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "pink",
        }
      }
      viewBox="0 0 24 24"
      width={42}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58a.49.49 0 00.12-.61l-1.92-3.32a.488.488 0 00-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54a.484.484 0 00-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58a.49.49 0 00-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
      />
    </svg>
  </div>,
  <div
    className="row"
  >
    <div
      className="col-7"
    >
      <svg
        className="fa-icon"
        fill="currentColor"
        height={98}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 496 512"
        width={98}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm200 248c0 22.5-3.9 44.2-10.8 64.4h-20.3c-4.3 0-8.4-1.7-11.4-4.8l-32-32.6c-4.5-4.6-4.5-12.1.1-16.7l12.5-12.5v-8.7c0-3-1.2-5.9-3.3-8l-9.4-9.4c-2.1-2.1-5-3.3-8-3.3h-16c-6.2 0-11.3-5.1-11.3-11.3 0-3 1.2-5.9 3.3-8l9.4-9.4c2.1-2.1 5-3.3 8-3.3h32c6.2 0 11.3-5.1 11.3-11.3v-9.4c0-6.2-5.1-11.3-11.3-11.3h-36.7c-8.8 0-16 7.2-16 16v4.5c0 6.9-4.4 13-10.9 15.2l-31.6 10.5c-3.3 1.1-5.5 4.1-5.5 7.6v2.2c0 4.4-3.6 8-8 8h-16c-4.4 0-8-3.6-8-8s-3.6-8-8-8H247c-3 0-5.8 1.7-7.2 4.4l-9.4 18.7c-2.7 5.4-8.2 8.8-14.3 8.8H194c-8.8 0-16-7.2-16-16V199c0-4.2 1.7-8.3 4.7-11.3l20.1-20.1c4.6-4.6 7.2-10.9 7.2-17.5 0-3.4 2.2-6.5 5.5-7.6l40-13.3c1.7-.6 3.2-1.5 4.4-2.7l26.8-26.8c2.1-2.1 3.3-5 3.3-8 0-6.2-5.1-11.3-11.3-11.3H258l-16 16v8c0 4.4-3.6 8-8 8h-16c-4.4 0-8-3.6-8-8v-20c0-2.5 1.2-4.9 3.2-6.4l28.9-21.7c1.9-.1 3.8-.3 5.7-.3C358.3 56 448 145.7 448 256zM130.1 149.1c0-3 1.2-5.9 3.3-8l25.4-25.4c2.1-2.1 5-3.3 8-3.3 6.2 0 11.3 5.1 11.3 11.3v16c0 3-1.2 5.9-3.3 8l-9.4 9.4c-2.1 2.1-5 3.3-8 3.3h-16c-6.2 0-11.3-5.1-11.3-11.3zm128 306.4v-7.1c0-8.8-7.2-16-16-16h-20.2c-10.8 0-26.7-5.3-35.4-11.8l-22.2-16.7c-11.5-8.6-18.2-22.1-18.2-36.4v-23.9c0-16 8.4-30.8 22.1-39l42.9-25.7c7.1-4.2 15.2-6.5 23.4-6.5h31.2c10.9 0 21.4 3.9 29.6 10.9l43.2 37.1h18.3c8.5 0 16.6 3.4 22.6 9.4l17.3 17.3c3.4 3.4 8.1 5.3 12.9 5.3H423c-32.4 58.9-93.8 99.5-164.9 103.1z"
        />
      </svg>
    </div>
    <div
      className="col-5 mt-5 text"
    >
      <h1
        className="d-block"
      />
      <p
        className="btn btn-dark"
      >
        chapters
      </p>
    </div>
    <h5
      className="text mt-2"
    >
      Chapter one
    </h5>
  </div>,
]
`;
